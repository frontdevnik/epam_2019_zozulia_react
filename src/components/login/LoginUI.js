import React from 'react';
import { Link } from 'react-router-dom';
import { Routes } from '../../constans';
import withTranslation from '../../hocs/withTranslation';
import styles from './login.module.scss';

function LoginUI(props) {
  const { loginUser, login_title, login_name, login_password, login_button, form_noAccount, form_title, link_to_registration, login_name_placeholder, login_password_placeholder } = props;

  return (
    <section className={styles.login}>
      <h1 className={styles.header}>{form_title}</h1>
      <form className={styles.form}>
        <h2 className={styles.formTitle}>{login_title}</h2>
        <label>
          {login_name}
          <input id="name" placeholder={login_name_placeholder} />
        </label>
        <label>
          {login_password}
          <input id="password" type="password" placeholder={login_password_placeholder} />
        </label>
        <p id="incorrect" className={styles.incorrect}></p>
        <button onClick={loginUser}>{login_button}</button>
        <p className={styles.register}>
          {form_noAccount}
          <Link to={Routes.REGISTRATION}> {link_to_registration}</Link>
        </p>
      </form>
    </section>
  );
};

const withTranslationWorlds = withTranslation(['login_title', 'login_name', 'login_password', 'login_button', 'form_noAccount', 'form_title', 'link_to_registration', 'login_name_placeholder', 'login_password_placeholder']);

export default withTranslationWorlds(LoginUI);
