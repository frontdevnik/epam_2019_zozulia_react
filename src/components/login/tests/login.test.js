import React from 'react';
import { shallow } from 'enzyme';
import LoginUI from '../LoginUI';
import styles from './login.module.scss';

fdescribe('<Login />', () => {
  let wrapper;

  const props = {
    loginUser: jest.fn(),
    login_title: 'Login Form',
    login_name: 'Input your name',
    login_password: 'Input your password',
    login_button: 'Login',
    form_noAccount: 'Don\'t have an account?',
    form_title: 'Movies',
    link_to_registration: 'Go to register page',
    login_name_placeholder: 'Input your name',
    login_password_placeholder: 'Input your password',
  }

  beforeEach(() => {
    wrapper = shallow(<LoginUI {...props} />).dive();
  })

  it('should render correctly', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render title from props', () => {
    expect(wrapper.find('h1').text()).toEqual(props.form_title);
    expect(wrapper.find('h1').hasClass(styles.header)).toBeTruthy();
  });

  it('should render form title from props', () => {
    expect(wrapper.find('h2').text()).toEqual(props.login_title);
    expect(wrapper.find('h2').hasClass(styles.formTitle)).toBeTruthy();
  });

  it('should render label name from props', () => {
    expect(wrapper.find('label').at(0).text()).toEqual(props.login_name);
  });

  it('should render label password from props', () => {
    expect(wrapper.find('label').at(1).text()).toEqual(props.login_password);
  });

  it('should render empty incorrect paragraph', () => {
    expect(wrapper.find('#incorrect').text()).toEqual('');
  });

  it('should invoke loginUser after button click', () => {
    const button = wrapper.find('button');
    button.simulate('click');
    expect(props.loginUser).toBeCalledTimes(1);
  });
});
