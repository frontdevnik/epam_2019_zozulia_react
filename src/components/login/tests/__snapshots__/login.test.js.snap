// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render correctly 1`] = `
"<section className=\\"login\\">
  <h1 className=\\"header\\">
    Movies
  </h1>
  <form className=\\"form\\">
    <h2 className=\\"formTitle\\">
      Login Form
    </h2>
    <label>
      Input your name
      <input id=\\"name\\" placeholder=\\"Input your name\\" />
    </label>
    <label>
      Input your password
      <input id=\\"password\\" type=\\"password\\" placeholder=\\"Input your password\\" />
    </label>
    <p id=\\"incorrect\\" className=\\"incorrect\\" />
    <button onClick={[Function: mockConstructor]}>
      Login
    </button>
    <p className=\\"register\\">
      Don&#39;t have an account?
      <Link to=\\"/registration\\">
         
        Go to register page
      </Link>
    </p>
  </form>
</section>"
`;
